src/appengine/libs/issue_management/jira/__init__.py

```diff
@@ -0,0 +1,200 @@
+    return self.jira_issue.fields.reporter
+
+  @reporter.setter
+  def reporter(self, new_reporter):
+    self.jira_issue.fields.reporter = new_reporter
+
+  @property
+  def is_open(self):
+    """Whether the issue is open."""
+    if self.jira_issue.resolution not in ['Closed', 'Done', 'Resolved']:
+      return True
+    return False
+
+  @property
+  def closed_time(self):
+    return self.jira_issue.fields.resolutiondate
+
+  @property
+  def status(self):
+    """The issue status."""
+    return self.jira_issue.fields.status
+
+  @status.setter
+  def status(self, new_status):
+    self.jira_issue.fields.status = new_status
+
+  @property
+  def body(self):
+    """The issue body."""
+    return self.jira_issue.fields.description
+
+  @body.setter
+  def body(self, new_body):
+    self.jira_issue.fields.description = new_body
+
+  @property
+  def assignee(self):
+    """The issue assignee."""
+    return self.jira_issue.fields.assignee
+
+  @assignee.setter
+  def assignee(self, new_assignee):
+    self.jira_issue.fields.assignee = new_assignee
+
+  @property
+  def ccs(self):
+    """The issue CC list."""
+    return self._ccs
+
+  @property
+  def labels(self):
+    """The issue labels list."""
+    return self._labels
+
+  @property
+  def components(self):
+    """The issue component list."""
+    return self._components
+
+  def save(self):
+    """Save the issue."""
+
+    for added in self._components.added:
+      self.components.add(added)
+    for removed in self._components.removed:
+      self.components.remove(removed)
+    self._components.reset_tracking()
+
+    for added in self._ccs.added:
+      self.ccs.add(added)
+    for removed in self._ccs.removed:
+      self.ccs.remove(removed)
+    self._ccs.reset_tracking()
+
+    for added in self._labels.added:
+      self.labels.add(added)
+    for removed in self._labels.removed:
+      self.labels.remove(removed)
+    self._labels.reset_tracking()
+
+    self.itm.save(self)
+
+
+class IssueTracker(issue_tracker.IssueTracker):
+  """Issue tracker interface."""
+
+  def __init__(self, itm):
+    self._itm = itm
+
+  @property
+  def project(self):
+    return self._itm.project_name
+
+  def new_issue(self):
+    jira_issue = self._itm.create()
+    wrapped_issue = Issue(self._itm, jira_issue)
+    return wrapped_issue
+
+  def get_issue(self, issue_id):
+    jira_issue = self._itm.get_issue(issue_id)
+    if not jira_issue:
+      return None
+
+    return Issue(self._itm, jira_issue)
+
+  def find_issues(self, keywords=None, only_open=False):
+    """Find issues."""
+    search_text = _get_search_text(keywords)
+    if only_open:
+      search_text += "AND resolution = Unresolved"
+    issues = self._itm.get_issues(search_text)
+    return [Issue(self._itm, issue) for issue in issues]
+
+  def issue_url(self, issue_id):
+    """Return the issue URL with the given ID."""
+    config = db_config.get()
+    url = config.jira_url + "/browse/" + str(issue_id)
```

Comment:
Please use single quotes for all strings that aren't docstrings (Also on line 166 and probably others).

Reply:
Will do